more_itertools/more.py

```diff
@@ -801,9 +801,13 @@ def interleave_longest(*iterables):
         >>> list(interleave_longest([1, 2, 3], [4, 5], [6, 7, 8]))
         [1, 4, 6, 2, 5, 7, 3, 8]
 
+    This function produces the same output as :func:`roundrobin`, but may
+    perform better for some inputs (in particular when the number of iterables
+    is large).
+
     """
     i = chain.from_iterable(zip_longest(*iterables, fillvalue=_marker))
-    return filter(lambda x: x is not _marker, i)
+    return (x for x in i if x is not _marker)
```

Comment:
Of course, this changes the result from a list to an iterable, but that’s how it’s specced, so it should be fine.

Reply:
FYI, the original didn't return a list due to [this import](https://github.com/erikrose/more-itertools/blob/821450b4d31979ae1f3bd88992e9401154260185/more_itertools/more.py#L22).

Reply:
Thank you for clarifying.

Reply:
Aha. Tricky!

Reply:
That’s what I get for reviewing on a handheld. ;-)